index.js

```diff
@@ -0,0 +1,133 @@
+'use strict'
+
+const { extractSocketDetails, protocolDecoder } = require('aedes-protocol-decoder')
+const http = require('http')
+const https = require('https')
+const http2 = require('http2')
+const net = require('net')
+const { Transform, PassThrough } = require('stream')
+const tls = require('tls')
+const WebSocket = require('ws')
+
+const defaultOptions = {
+  ws: false,
+  https: false,
+  http2: false,
+  tls: false,
+  trustProxy: false,
+  serverFactory: null,
+  protocolDecoder: protocolDecoder,
+  extractSocketDetails: extractSocketDetails
+}
+
+const createServer = (options, aedesHandler) => {
```

Comment:
I was thinking about the signature. What about use: `createServer (broker, options)` ? I think it is more readable and user friendly

Than inside the code you can use `broker.handle` but also in future we could hack something else with the broker reference :)

Reply:
Good idea!